v4/weaviate/graphql/get.go

```diff
@@ -32,168 +36,175 @@ type GetBuilder struct {
 	withHybrid           *HybridArgumentBuilder
 }
 
-// WithClassName that should be queried
-func (gb *GetBuilder) WithClassName(name string) *GetBuilder {
-	gb.className = name
-	return gb
+// ClassName that should be queried
+func NewQueryClassBuilder(className string) *ClassBuilder {
```

Comment:
This is a breaking change, right? And all users would need to update their implementation?

Wouldn't be a big fan of this. Can we keep the old behavior and add the possiblity to have multiple classes in addition? @parkerduckworth what do you think?

Reply:
that looks much better :)

I noticed that it duplicates a lot of the code for GetBuilder, I think we can improve that.

How about doing:
- in get.go:
```
type GetBuilder struct {
	connection rest
        GetBuilderBase
}

type GetBuilderBase struct{
	className  string
        ..other members...
}
```
- in multiclassbuilder.go
```
type MultiClassBuilder struct {
	connection    rest
	classBuilders map[string]*GetBuilderMulti
}

type GetBuilderMulti struct {
        GetBuilderBase
}

...additional member functions....
```
(not tested, just typed here)

A few more tests would be great, though

Reply:
I've tried to incorporate your suggestions but in a slightly different way. Let me know your thoughts. If this looks good, will add tests for `MultiClassBuilder `.

Reply:
Hi,

now we have the problem again, that we have a breaking change :/ Sorry, didn't think of that.

New proposal:
How about we leave the `GetBuilder` class like it is. And then have the following?
```
type MultiClassBuilder struct {
	connection    rest
	classBuilders map[string]*GetBuilder
}

NewMultiClassBuilder(connection rest){
return MultiClassBuilder{....}
}

func (m *MultiClassBuilder) AddClass(className string) *GetBuilder{
  m.classBuilders[className] = GetBuilder{conn: m.connection, className: className}
  return m.classBuilders[className]
}

func (m *MultiClassBuilder) ModifyClass(className string) *GetBuilder{
  return m.classBuilders[className]
}
```

Then we're backwards compatible and have a nice interface to add multiple classes. What do you think? Hope I didn't miss anything :)

Instead of returning the GetBuilder, we could also go for something that can be chained with multiple classes

Reply:
Hey @dirkkul ,

Sorry I couldn't understand how this is a breaking change. The only thing we are changing is the underlying structure of `client.GraphQL().Get()`, the fields defined on this weren't even earlier exposed/accessible and all the methods are still defined as they were earlier.

It would be great if you can add an example demonstrating code breaking.

Reply:
Hi,

before users would write

```
		builder := GetBuilder{
			connection:           conMock,
			..other options...
		}
```

now they need to write the following and update all existing code.
```
GetBuilder{
			connection: conMock,
			builderBase: builderBase{
				includesFilterClause: false,
			},
		}
```

Or did I miss something? At least you updated a few tests that use the GetBuilder.

Reply:
I think you misunderstood, the file that you're referring for changes includes unit tests. Yes that is changed. But end user actually won't be using it this way, right? 

The way users will be using isn't changes. Even before users were doing this and even after this change they will be doing :

`
client.GraphQL().Get().WithClassName("Pizza").WithFields(name).Do(context.Background())
`

You can refer to weaviate-go-client/v4/test/grapgql/graphql_test.go file for more example.

Reply:
@dirkkul, I thing this would help you understand my thoughts,

You are only half-correct when you say users can write this : 
```
builder := GetBuilder{
	connection:           conMock,  
	..other options...  
}
```
Outside `weaviate-go-client/v4/weaviate`, users will only be able to write this :
```
builder := graphql.GetBuilder{}
```
Outside the weaviate pkg, users won't be able to set `connection` or any other field of `GetBuilder` structure.

Let's extent this argument, assuming user is writing something like this 
```
bd := graphql.GetBuilder{}
bd.WithClassName("Pizza").WithAsk(...)
```
and since there is no way to pass variable `bd` as argument to some function and execute the get query, the only way user can execute this query is by doing this : 
```
bd.Do(context.Background())
```
and this is gonna return some error, since `connection` field isn't set. So logically speaking there is no point of getting a `graphql.GetBuilder` variable without using `client.GraphQL().Get()` method.

So yes, you are correct that user will have to re-write/update existing code if they are declaring `graphql.GetBuilder` variable without using `client.GraphQL().Get()` method, but I believe this is very less likely that someone will be doing it the wrong way, since there is no point of doing it.

Please correct me, if I am wrong.